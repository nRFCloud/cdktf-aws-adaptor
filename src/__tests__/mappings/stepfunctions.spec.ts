import { SfnStateMachine } from "@cdktf/provider-aws/lib/sfn-state-machine";
import { CfnStateMachine } from "aws-cdk-lib/aws-stepfunctions";
import { setupJest } from "cdktf/lib/testing/adapters/jest.js";
import { describe } from "vitest";
import { synthesizeElementAndTestStability } from "../helpers.js";

setupJest();

describe("Step Functions mappings", () => {
    describe("AWS:StepFunctions:StateMachine", () => {
        it("Should translate with String definition", () => {
            synthesizeElementAndTestStability(
                CfnStateMachine,
                {
                    definitionString: "{\"StartAt\": \"HelloWorld\", \"States\": {}}",
                    stateMachineName: "name",
                    tags: [{
                        key: "key",
                        value: "value",
                    }],
                    stateMachineType: "STANDARD",
                    roleArn: "roleArn",
                    loggingConfiguration: {
                        level: "ALL",
                        destinations: [{
                            cloudWatchLogsLogGroup: {
                                logGroupArn: "logGroupArn",
                            },
                        }],
                        includeExecutionData: true,
                    },
                    tracingConfiguration: {
                        enabled: true,
                    },
                },
                SfnStateMachine,
                {
                    name: "name",
                    definition: "\${join(\"\", [\"{\\\"StartAt\\\": \\\"HelloWorld\\\", \\\"States\\\": {}}\"])}",
                    roleArn: "roleArn",
                    loggingConfiguration: {
                        level: "ALL",
                        logDestination: "logGroupArn:*",
                        includeExecutionData: true,
                    },
                    type: "STANDARD",
                    tags: {
                        key: "value",
                    },
                    tracingConfiguration: {
                        enabled: true,
                    },
                },
            );
        });

        it("Should translate with S3 definition", () => {
            synthesizeElementAndTestStability(
                CfnStateMachine,
                {
                    definitionS3Location: {
                        bucket: "bucket",
                        key: "key",
                    },
                    stateMachineName: "name",
                    tags: [{
                        key: "key",
                        value: "value",
                    }],
                    stateMachineType: "STANDARD",
                    roleArn: "roleArn",
                    loggingConfiguration: {
                        level: "ALL",
                        destinations: [{
                            cloudWatchLogsLogGroup: {
                                logGroupArn: "logGroupArn",
                            },
                        }],
                        includeExecutionData: true,
                    },
                    tracingConfiguration: {
                        enabled: true,
                    },
                    encryptionConfiguration: {
                        kmsDataKeyReusePeriodSeconds: 600,
                        kmsKeyId: "123",
                        type: "Test",
                    },
                },
                SfnStateMachine,
                {
                    name: "name",
                    definition: "${data.aws_s3_bucket_object.resource_resource-definition_AA6652B4.body}",
                    roleArn: "roleArn",
                    loggingConfiguration: {
                        level: "ALL",
                        logDestination: "logGroupArn:*",
                        includeExecutionData: true,
                    },
                    type: "STANDARD",
                    tags: {
                        key: "value",
                    },
                    tracingConfiguration: {
                        enabled: true,
                    },
                    encryptionConfiguration: {
                        kmsDataKeyReusePeriodSeconds: 600,
                        kmsKeyId: "123",
                        type: "Test",
                    },
                },
            );
        });

        it("Should translate with Object definition", () => {
            synthesizeElementAndTestStability(
                CfnStateMachine,
                {
                    definition: {
                        StartAt: "HelloWorld",
                        States: {},
                    },
                    stateMachineName: "name",
                    tags: [{
                        key: "key",
                        value: "value",
                    }],
                    stateMachineType: "STANDARD",
                    roleArn: "roleArn",
                    loggingConfiguration: {
                        level: "ALL",
                        destinations: [{
                            cloudWatchLogsLogGroup: {
                                logGroupArn: "logGroupArn",
                            },
                        }],
                        includeExecutionData: true,
                    },
                    tracingConfiguration: {
                        enabled: true,
                    },
                },
                SfnStateMachine,
                {
                    name: "name",
                    definition: "\${jsonencode({\"StartAt\" = \"HelloWorld\", \"States\" = {}})}",
                    roleArn: "roleArn",
                    loggingConfiguration: {
                        level: "ALL",
                        logDestination: "logGroupArn:*",
                        includeExecutionData: true,
                    },
                    type: "STANDARD",
                    tags: {
                        key: "value",
                    },
                    tracingConfiguration: {
                        enabled: true,
                    },
                },
            );
        });

        it("Should translate with definition substitutions", () => {
            synthesizeElementAndTestStability(
                CfnStateMachine,
                {
                    definitionS3Location: {
                        bucket: "bucket",
                        key: "key",
                    },
                    definitionSubstitutions: {
                        "MyFunction": "arn:aws:lambda:us-east-1:123456789012:function:my-function:1",
                        "Task": "TaskType",
                    },
                    stateMachineName: "name",
                    tags: [{
                        key: "key",
                        value: "value",
                    }],
                    stateMachineType: "STANDARD",
                    roleArn: "roleArn",
                    loggingConfiguration: {
                        level: "ALL",
                        destinations: [{
                            cloudWatchLogsLogGroup: {
                                logGroupArn: "logGroupArn",
                            },
                        }],
                        includeExecutionData: true,
                    },
                    tracingConfiguration: {
                        enabled: true,
                    },
                },
                SfnStateMachine,
                {
                    name: "name",
                    definition:
                        "${replace(replace(data.aws_s3_bucket_object.resource_resource-definition_AA6652B4.body, \"$${MyFunction}\", \"arn:aws:lambda:us-east-1:123456789012:function:my-function:1\"), \"$${Task}\", \"TaskType\")}",
                    roleArn: "roleArn",
                    loggingConfiguration: {
                        level: "ALL",
                        logDestination: "logGroupArn:*",
                        includeExecutionData: true,
                    },
                    type: "STANDARD",
                    tags: {
                        key: "value",
                    },
                    tracingConfiguration: {
                        enabled: true,
                    },
                },
            );
        });
    });
});
