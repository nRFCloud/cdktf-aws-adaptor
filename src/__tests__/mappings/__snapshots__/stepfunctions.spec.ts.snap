// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Step Functions mappings > AWS:StepFunctions:StateMachine > Should translate with Object definition 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_sfn_state_machine\\": {
      \\"resource_22C949BF\\": {
        \\"definition\\": \\"\${jsonencode({\\\\\\"StartAt\\\\\\" = \\\\\\"HelloWorld\\\\\\", \\\\\\"States\\\\\\" = {}})}\\",
        \\"logging_configuration\\": {
          \\"include_execution_data\\": true,
          \\"level\\": \\"ALL\\",
          \\"log_destination\\": \\"logGroupArn:*\\"
        },
        \\"name\\": \\"name\\",
        \\"role_arn\\": \\"roleArn\\",
        \\"tags\\": {
          \\"key\\": \\"value\\"
        },
        \\"tracing_configuration\\": {
          \\"enabled\\": true
        },
        \\"type\\": \\"STANDARD\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${join(\\\\\\"\\\\\\", [aws_sfn_state_machine.resource_22C949BF.arn, \\\\\\",\\\\\\", aws_sfn_state_machine.resource_22C949BF.name, \\\\\\",\\\\\\", aws_sfn_state_machine.resource_22C949BF.revision_id])}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Step Functions mappings > AWS:StepFunctions:StateMachine > Should translate with S3 definition 1`] = `
"{
  \\"data\\": {
    \\"aws_s3_bucket_object\\": {
      \\"resource_resource-definition_AA6652B4\\": {
        \\"bucket\\": \\"bucket\\",
        \\"key\\": \\"key\\"
      }
    }
  },
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_sfn_state_machine\\": {
      \\"resource_22C949BF\\": {
        \\"definition\\": \\"\${data.aws_s3_bucket_object.resource_resource-definition_AA6652B4.body}\\",
        \\"encryption_configuration\\": {
          \\"kms_data_key_reuse_period_seconds\\": 600,
          \\"kms_key_id\\": \\"123\\",
          \\"type\\": \\"Test\\"
        },
        \\"logging_configuration\\": {
          \\"include_execution_data\\": true,
          \\"level\\": \\"ALL\\",
          \\"log_destination\\": \\"logGroupArn:*\\"
        },
        \\"name\\": \\"name\\",
        \\"role_arn\\": \\"roleArn\\",
        \\"tags\\": {
          \\"key\\": \\"value\\"
        },
        \\"tracing_configuration\\": {
          \\"enabled\\": true
        },
        \\"type\\": \\"STANDARD\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${join(\\\\\\"\\\\\\", [aws_sfn_state_machine.resource_22C949BF.arn, \\\\\\",\\\\\\", aws_sfn_state_machine.resource_22C949BF.name, \\\\\\",\\\\\\", aws_sfn_state_machine.resource_22C949BF.revision_id])}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Step Functions mappings > AWS:StepFunctions:StateMachine > Should translate with String definition 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_sfn_state_machine\\": {
      \\"resource_22C949BF\\": {
        \\"definition\\": \\"\${join(\\\\\\"\\\\\\", [\\\\\\"{\\\\\\\\\\\\\\"StartAt\\\\\\\\\\\\\\": \\\\\\\\\\\\\\"HelloWorld\\\\\\\\\\\\\\", \\\\\\\\\\\\\\"States\\\\\\\\\\\\\\": {}}\\\\\\"])}\\",
        \\"logging_configuration\\": {
          \\"include_execution_data\\": true,
          \\"level\\": \\"ALL\\",
          \\"log_destination\\": \\"logGroupArn:*\\"
        },
        \\"name\\": \\"name\\",
        \\"role_arn\\": \\"roleArn\\",
        \\"tags\\": {
          \\"key\\": \\"value\\"
        },
        \\"tracing_configuration\\": {
          \\"enabled\\": true
        },
        \\"type\\": \\"STANDARD\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${join(\\\\\\"\\\\\\", [aws_sfn_state_machine.resource_22C949BF.arn, \\\\\\",\\\\\\", aws_sfn_state_machine.resource_22C949BF.name, \\\\\\",\\\\\\", aws_sfn_state_machine.resource_22C949BF.revision_id])}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Step Functions mappings > AWS:StepFunctions:StateMachine > Should translate with definition substitutions 1`] = `
"{
  \\"data\\": {
    \\"aws_s3_bucket_object\\": {
      \\"resource_resource-definition_AA6652B4\\": {
        \\"bucket\\": \\"bucket\\",
        \\"key\\": \\"key\\"
      }
    }
  },
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_sfn_state_machine\\": {
      \\"resource_22C949BF\\": {
        \\"definition\\": \\"\${replace(replace(data.aws_s3_bucket_object.resource_resource-definition_AA6652B4.body, \\\\\\"$\${MyFunction}\\\\\\", \\\\\\"arn:aws:lambda:us-east-1:123456789012:function:my-function:1\\\\\\"), \\\\\\"$\${Task}\\\\\\", \\\\\\"TaskType\\\\\\")}\\",
        \\"logging_configuration\\": {
          \\"include_execution_data\\": true,
          \\"level\\": \\"ALL\\",
          \\"log_destination\\": \\"logGroupArn:*\\"
        },
        \\"name\\": \\"name\\",
        \\"role_arn\\": \\"roleArn\\",
        \\"tags\\": {
          \\"key\\": \\"value\\"
        },
        \\"tracing_configuration\\": {
          \\"enabled\\": true
        },
        \\"type\\": \\"STANDARD\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${join(\\\\\\"\\\\\\", [aws_sfn_state_machine.resource_22C949BF.arn, \\\\\\",\\\\\\", aws_sfn_state_machine.resource_22C949BF.name, \\\\\\",\\\\\\", aws_sfn_state_machine.resource_22C949BF.revision_id])}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;
